### ISSUE TITLE
New check | consider using any/all

### ISSUE THREAD

**Issue by sushobhit27** [PyCQA/pylint#2227]
### Steps to reproduce
1. cat sample.py
```python
a = True
b = True
c = True

if a or b or c:
    print('either of a,b, c is true')

if any([a, b, c]):
    print('atleast something is true')

``` 
2. $ pylint sample.py

### Current behavior
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

### Expected behavior
There should be a refactoring message at line 5 stating that it could be simplified as line 8.

### pylint --version output
pylint 2.0.0.dev1
astroid 2.0.0.dev3
Python 3.6.4 (default, Jan  7 2018, 15:53:53)
[GCC 6.4.0]

**Comment by PCManticore** [comment]
Sounds good to me!

**Comment by AWhetter** [comment]
My preference here is that this isn't a default check. I don't like that you have to call a function and construct a list/tuple only to do something that the language has a keyword for.
It also masks complexity. Line 8 would artificially reduce the complexity number by one because there's only a single "if" vs an "if" with two "or"s. Perhaps that's a problem with our (and flake8's) complexity checker, but I still find that a developers intentions are clearer when using "or".

**Comment by sushobhit27** [comment]
@AWhetter IMO Till there are 2-3 operands in or/and, simply using operators is fine but when there are more operands, using any/all looks more concise. Since we have option to disable any message, its better to have some functionality and selectively enable/disable it rather than not having a new check altogther.
However you and @PCManticore can decide whether to have this check or not.

**Comment by PCManticore** [comment]
@AWhetter convinced me. Let's close this @sushobhit27 and sorry for already writing a checker for it!
